config/supermicro.go

```diff
@@ -114,13 +125,208 @@ func (cm *supermicroVendorConfig) Marshal() (string, error) {
+		for _, s := range menu.Settings {
+			switch s.Name {
+			// We want to drop this list of settings
+			case "NewSetupPassword", "NewSysPassword", "OldSetupPassword", "OldSysPassword":
+			// All others get normalized
+			default:
+				var k, v string
+				k, v, err = normalizeSetting(s)
+
+				if err != nil {
+					return
+				}
+
+				biosConfig[k] = v
+			}
+		}
+	}
+
+	return
+}
+
+func normalizeSetting(s *supermicroBiosCfgSetting) (k, v string, err error) {
+	switch s.Type {
+	case "CheckBox":
+		k = normalizeName(s.Name)
+		v = normalizeValue(k, s.CheckedStatus)
+	case "Option":
+		k = normalizeName(s.Name)
+		v = normalizeValue(k, s.SelectedOption)
+	case "Password":
+		k = normalizeName(s.Name)
+		v = ""
+	case "Numeric":
+		k = normalizeName(s.Name)
+		v = normalizeValue(k, s.NumericValue)
+	default:
+		err = UnknownSettingType(s.Type)
+		return
+	}
+
+	return
+}
+
+func normalizeName(k string) string {
+	switch k {
+	case "CpuMinSevAsid":
+		return "amd_sev"
+	case "BootMode", "Boot mode select":
+		return "boot_mode"
+	case "IntelTxt":
+		return "intel_txt"
+	case "Software Guard Extensions (SGX)":
+		return "intel_sgx"
+	case "SecureBoot", "Secure Boot":
+		return "secure_boot"
+	case "Hyper-Threading", "Hyper-Threading [ALL]", "LogicalProc":
+		return "smt"
+	case "SriovGlobalEnable":
+		return "sr_iov"
+	case "TpmSecurity", "Security Device Support":
+		return "tpm"
+	default:
+		// When we don't normalize the key prepend "raw:"
+		return "raw:" + k
+	}
+}
+
+func normalizeBootMode(v string) string {
+	switch strings.ToLower(v) {
+	case "legacy":
+		return "BIOS"
+	default:
+		return strings.ToUpper(v)
+	}
+}
+
+func normalizeValue(k, v string) string {
+	if k == "boot_mode" {
+		return normalizeBootMode(v)
+	}
+
+	switch strings.ToLower(v) {
+	case "disable":
+		return disabledValue
+	case "disabled":
+		return disabledValue
+	case "enable":
+		return enabledValue
+	case "enabled":
+		return enabledValue
+	case "off":
+		return disabledValue
+	case "on":
+		return enabledValue
+	default:
+		return v
+	}
+}
+
 // Generic config options
 
-func (cm *supermicroVendorConfig) EnableTPM() {
-	cm.Raw("  Security Device Support", "Enable", []string{"Trusted Computing"})
-	cm.Raw("  SHA-1 PCR Bank", "Enabled", []string{"Trusted Computing"})
+func (cm *supermicroVendorConfig) BootMode(mode string) error {
+	switch strings.ToUpper(mode) {
+	case "LEGACY", "UEFI", "DUAL":
+		cm.Raw("Boot mode select", strings.ToUpper(mode), []string{"Boot"})
+	default:
+		return InvalidBootModeOption(strings.ToUpper(mode))
+	}
+
+	return nil
 }
 
-func (cm *supermicroVendorConfig) EnableSRIOV() {
-	cm.Raw("SR-IOV Support", "Enabled", []string{"Advanced", "PCIe/PCI/PnP Configuration"})
+func (cm *supermicroVendorConfig) BootOrder(mode string) error {
+	switch strings.ToUpper(mode) {
+	case "LEGACY":
+		cm.Raw("Legacy Boot Option #1", "Hard Disk", []string{"Boot"})
+		cm.Raw("Legacy Boot Option #2", "Network", []string{"Boot"})
+		for i := 3; i < 8; i++ {
```

Comment:
what is 3, 8 here exactly, could you define them as consts or add some comments on why those counts specifically